source/connect/tls.txt

```diff
@@ -0,0 +1,326 @@
+     - ``String``
+     - The certificate file path used to verify the connection to a MongoDB 
+       deployment.
+   * - ``:ssl_cert_object``
+     - ``OpenSSL::X509::Certificate``
+     - The certificate object used to verify the connection to a MongoDB 
+       deployment. 
+   * - ``:ssl_cert_string``
+     - ``String``
+     - A string containing the PEM-encoded certificate used to verify the connection to 
+       a MongoDB deployment.
+
+TLS Private Key
+~~~~~~~~~~~~~~~
+
+You can provide one of the following options to specify the TLS private key:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 30 50
+
+   * - Option Name
+     - Data/Object Type Accepted
+     - Description
+   * - ``:ssl_key``
+     - ``String``
+     - The private keyfile path used to verify the connection to a MongoDB 
+       deployment.
+   * - ``:ssl_key_object``
+     - ``OpenSSL::PKey``
+     - The private key object used to verify the connection to a MongoDB 
+       deployment.
+   * - ``:ssl_key_pass_phrase``
+     - ``String``
+     - A passphrase for the private key.
+   * - ``:ssl_key_string``
+     - ``String``
+     - A string containing the PEM-encoded private key used to verify the 
+       connection to a MongoDB deployment.
+
+TLS CA Certificate
+~~~~~~~~~~~~~~~~~~
+
+You can provide one of the following options to specify the TLS CA certificate:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 30 50
+
+   * - Option Name
+     - Data/Object Type Accepted
+     - Description
+   * - ``:ssl_ca_cert``
+     - ``String``
+     - The file path containing concatenated CA certificates used to validate certificates 
+       passed from the MongoDB deployment to the client.
+   * - ``:ssl_ca_cert_object``
+     - ``Array<OpenSSL::X509::Certificate>``
+     - An array of objects representing the CA certificates used 
+       to validate certificates passed from the MongoDB deployment to the client.
+   * - ``:ssl_ca_cert_string``
+     - ``String``
+     - A string containing one PEM-encoded CA certificate used to validate certificates 
+       passed from the MongoDB deployment to the client.
+
+Modify the TLS Context
+----------------------
+
+If your TLS configuration requires customization, you can set TLS context hooks
+by adding a native Ruby ``Proc`` object to the ``Mongo.tls_context_hooks`` 
+array. Add the ``Proc`` object to the array before you create any ``Mongo::Client`` 
+instances. 
+
+The following code example enables the AES256-SHA cipher as the only cipher 
+to be used for TLS:
+
+.. literalinclude:: /includes/connect/tls.rb
+    :language: ruby
+    :dedent:
+    :start-after: start-modify-context
+    :end-before: end-modify-context
+
+The {+driver-short+} TLS context options are based on native Ruby handling of SSL. To 
+learn more about the TLS context options available, see the Ruby
+documentation for 
+`SSLContext <https://ruby-doc.org/stdlib-2.7.0/libdoc/openssl/rdoc/OpenSSL/SSL/SSLContext.html>`__.
```

Comment:
For technical reviewer, is this an appropriate link to put for reference? I wasn't able to find much information on ``context`` options in our docs. Thanks!

Reply:
Ruby 2.7 is EOL, so that link is probably not ideal. Linking out to the Ruby docs is otherwise probably fine, we just need to make sure the links remain relevant.

Here's the link to the same class, for Ruby 3.2.6: https://ruby-doc.org/3.2.6/exts/openssl/OpenSSL/SSL/SSLContext.html